apollo-ios/Sources/Apollo/CacheWriteInterceptor.swift

```diff
@@ -37,7 +32,11 @@ public struct CacheWriteInterceptor: ApolloInterceptor {
     request: HTTPRequest<Operation>,
     response: HTTPResponse<Operation>?,
     completion: @escaping (Result<GraphQLResult<Operation.Data>, any Error>) -> Void) {
-    
+
+    guard !chain.isCancelled else {
```

Comment:
I've moved this earlier in the `interceptAsync` logic because it makes sense that if the chain has already been cancelled you would not want to receive any response, nor should the interceptor chain proceed forward.